cypress/integration/plugins/dashboards-flow-framework/create_workflow_spec.js

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  FF_URL,
+  FF_FIXTURE_BASE_PATH,
+  MODEL_PARAMETERS,
+  WORKFLOW_DETAIL_URL_SEGMENT,
+} from '../../../utils/constants';
+import createConnectorBody from '../../../fixtures/plugins/dashboards-flow-framework/create_connector.json';
+import registerModelBody from '../../../fixtures/plugins/dashboards-flow-framework/register_model.json';
+import { getLastPathSegment } from '../../../utils/plugins/dashboards-flow-framework/helpers';
+
+describe('Creating Workflows Using Various Methods', () => {
+  before(() => {
+    cy.createConnector(createConnectorBody)
+      .then((connectorResponse) => {
+        MODEL_PARAMETERS.CONNECTOR_ID = connectorResponse.connector_id;
+        return cy.registerModel({
+          body: {
+            ...registerModelBody,
+            connector_id: MODEL_PARAMETERS.CONNECTOR_ID,
+            function_name: 'remote',
+          },
+        });
+      })
+      .then((modelResponse) => {
+        MODEL_PARAMETERS.MODEL_ID = modelResponse.model_id;
+        return cy.deployModel(MODEL_PARAMETERS.MODEL_ID);
+      });
+  });
+
+  beforeEach(() => {
+    cy.visit(FF_URL.WORKFLOWS_NEW);
+    cy.url().should('include', getLastPathSegment(FF_URL.WORKFLOWS_NEW));
+  });
+
+  it('create workflow using import', () => {
+    cy.getElementByDataTestId('importWorkflowButton')
+      .should('be.visible')
+      .click();
+    cy.contains('Import a workflow (JSON/YAML)').should('be.visible');
+    const filePath =
+      'cypress/fixtures/' +
+      FF_FIXTURE_BASE_PATH +
+      'import_semantic_search.json';
+    cy.get('input[type=file]').selectFile(filePath);
+    cy.getElementByDataTestId('importJSONButton').should('be.visible').click();
+    cy.wait(5000);
+    cy.get('.euiFieldSearch').focus();
+    cy.get('.euiFieldSearch').type('semantic_search_1{enter}');
+    cy.contains('semantic_search_1');
+    cy.get('.euiTableRow').should('have.length.greaterThan', 0);
+    cy.get('.euiTableRow').first().find('button.euiButtonIcon--danger').click();
+    cy.contains('The workflow will be permanently deleted.').should('exist');
+    cy.getElementByDataTestId('deleteWorkflowButton')
+      .should('be.visible')
+      .click();
+  });
+  it('Workflow Creation with Improper Import File', () => {
+    cy.getElementByDataTestId('importWorkflowButton')
+      .should('be.visible')
+      .click();
+    cy.contains('Import a workflow (JSON/YAML)').should('be.visible');
+    const filePath =
+      'cypress/fixtures/' + FF_FIXTURE_BASE_PATH + 'semantic_search_query.json';
+    cy.get('input[type=file]').selectFile(filePath);
+    cy.contains('The uploaded file is not a valid workflow').should(
+      'be.visible'
+    );
+  });
+
+  it('create workflow using Semantic Search template', () => {
+    cy.contains('h2', 'Semantic Search', { timeout: 120000 })
```

Comment:
We should avoid extreme timeout lengths if possible. `beforeEach` is just changing the URL and not performing any expensive operation, and should only take a few seconds max. Suggest to lower to maybe 10-20s maximum. The main reason being, if there is some issue and it _does_ not load, it will take significantly longer for the overall workflow to fail. Especially once hooked into the distribution test workflows, we don't want to add significant run times for those.

Also, may be helpful to move this timeout into a constant for consistency.